### DISCUSSION TITLE
Docker from within Actions

### DISCUSSION THREAD

**Discussion by ThomasAdam** [community/community#25962] [Actions]
<p>Hi all,</p>
<p>At $DAYJOB, we’re looking to make use of Github Actions as our CI/CD flow.  To do what we need to do, we’re making use of Docker to build our software across our repositories.</p>
<p>At the moment, we have a set of actions in our <code>~/.github/workflows/build.yml</code> file:</p>
<pre><code class="lang-auto"> steps:
    - uses: actions/checkout@v2
      with:
        repo-token: ${{ secrets.READ_TOKEN }}
    - name: Docker Login
      run: docker login docker.pkg.github.com -u $GITHUB_ACTOR -p ${{ secrets.READ_TOKEN }}
    - name: Pulling docker image
      run: docker pull docker.pkg.github.com/repo:latest
    - name: Build Package
      run: 'docker build -t package .'
</code></pre>
<p>The above works, but we have some concerns:</p>
<ul>
<li>This environment in Docker is self-contained – there’s no way of referencing anything outside of it, and as such we cannot therefore make use of third-party integrations, as we can’t access anything from within the Docker container;</li>
<li>We cannot make use of <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontainer" rel="nofollow noopener">https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontainer</a> since the <code>container:</code> directive only seems to be applicable to container images hosted on either Docker Hub or some other publicly-hosted site.</li>
</ul>
<p>We’re reluctant to want to add all of our deployment logic within the Docker container.  I cannot for the life of me believe that the <code>container:</code> block is only accessible for public Docker images.  Why wouldn’t it be the case that the <code>container:</code> block can access images stored on Github’s own Package site?</p>
<p>Without the ability to reference Github’s own package store via a <code>container:</code> block, essentually means that we cannot therefore set up mounted volumes and other things, which therefore makes our life slightly more difficult, and we’d like to avoid this.</p>
<p>We want to be able to mount volumes from these Docker containers, as the documentation suggests.  Without Github supporting accessing its own Package stores, this seems nonsensical.</p>
<p>Any thoughts much appreciated.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/thomasadam">@ThomasAdam</a>,</p>
<p>Sorry you experienced this.</p>
<p>Currently it’s not supported to pull packages from github packages for contianer image, this has been already raised as an internal feature_request, please refer to the similar <a href="https://github.community/t/docker-pull-from-public-github-package-registry-fail-with-no-basic-auth-credentials-error/16358/12">ticket</a> for the details.</p>
<p>Thanks</p>
